pkg/controller/compute/kubernetes/cluster.go

```diff
@@ -232,8 +233,10 @@ func (r *Reconciler) _bind(instance *computev1alpha1.KubernetesCluster) (reconci
 	instance.Status.SetBound()
 	instance.Status.CredentialsSecretRef = corev1.LocalObjectReference{Name: secret.Name}
 	// update conditions
-	instance.Status.UnsetAllConditions()
-	instance.Status.SetCondition(corev1alpha1.NewCondition(corev1alpha1.Ready, "", ""))
+	if !instance.Status.IsReady() {
+		instance.Status.UnsetAllConditions()
+		instance.Status.SetCondition(corev1alpha1.NewCondition(corev1alpha1.Ready, "", ""))
```

Comment:
can this just use `instance.Status.SetReady()` instead?

Reply:
totally, good catch.